### DISCUSSION TITLE
X96 X4 Gigabit - Network did not work

### DISCUSSION THREAD

**Discussion by Vanngocan** [devmfc/debian-on-amlogic#13] [Q&A]
Hello. Thank you for your great job!

I installed debian and ubuntu on X96 X4 Gigabit tv box, with 4GB RAM. However, the network did not work. I plugged the LAN cable to the box and nothing happened.

Could you please explain how to make network work on this box?

**Comment by devmfc** [comment]
Which image version did you use? Have you tried another one already. What box did you configure in boot.config? How do you know you really have a gigabit ethernet version? Did you try a 100MB version? Can you post a dmesg boot log? What is the output of `ethtool eth0`

Sorry my crystal ball is broken ;-)

**Reply by Vanngocan** [reply]
I am sorry, I am joining a trip so that I could not touch on my tvbox at this moment.

> Which image version did you use?

I used image: Devmfc_Debian-Bookworm_6.6.1-meson64_Minimal-23.11.09.img.xz.

> Have you tried another one already.

Yes, I tried Ubuntu image, and the result was the same.

> What box did you configure in boot.config?

My config was:
box=x96x4_gigabit
#root=/dev/sda2
root=/dev/mmcblk0p2

> How do you know you really have a gigabit ethernet version? Did you try a 100MB version?

I know this is gigabit version because I tested the LAN speed on Android. I also tried 100MB (box=x96x4), but the result was the same.

> Can you post a dmesg boot log? What is the output of ethtool eth0

Because I am outside now, I will send the log when I come back next week.

**Comment by NotDead** [comment]
Hello, I seem to have the same problem, on similar hardware: generic s905x4_gigabit dtb(vontar x4), "ip r" is empty if wi-fi is not connected.
[dmesg_without_network.log](https://github.com/devmfc/debian-on-amlogic/files/13427333/dmesg_without_network.log)
[ifconf_without_net.log](https://github.com/devmfc/debian-on-amlogic/files/13427381/ifconf_without_net.log)
[ethtool_with_net.log](https://github.com/devmfc/debian-on-amlogic/files/13438447/ethtool_with_net.log)

Now I'm on 6.6.1 debian(ubuntu is the same), also tried v6.1.57, booting from 2Gb microsd card
boot.config:
```
box=s905x4_generic_gigabit
#root=/dev/sda2
root=/dev/mmcblk0p2
```

If I connect to wi-fi, edit "/etc/systemd/network/10-eth0.network" and set a static address, I can connect to that address

**Reply by devmfc** [reply]
Thanks for the logs.
Two things stand out . All my images assume there is a rtl8211F ethernet PHY in case of gigabit. I think your box has something else.
The other thing is that your ethernet is up, but just doesn't get an IP address. Don't you have DHCP on your network? If not, you will have to set a static IP in */etc/systemd/network/10-eth0.network* 
Can you give the output of  `cat /sys/bus/mdio_bus/devices/*/{phy_interface,phy_id}|xargs`?  and can you try `dhclient eth0` and see if you do have an IP address after?

**Reply by NotDead** [reply]
Yes, I have DHCP but it only works for wi-fi on this board, the network with cable doesn't work.
as I removed static conf and restarted networkd, eth0 loose IPv4 run ```dhclient eth0``` and nothing is changed
Right after I deleted the static conf and restarted networkd, eth0 lost its IPv4 address, I ran ```dhclient eth0``` and nothing has changed

also tried ```dhclient -4 -v eth0```
```bash
Internet Systems Consortium DHCP Client 4.4.3-P1
Copyright 2004-2022 Internet Systems Consortium.
All rights reserved.
For info, please visit https://www.isc.org/software/dhcp/

Listening on LPF/eth0/00:ea:22:24:16:34
Sending on   LPF/eth0/00:ea:22:24:16:34
Sending on   Socket/fallback
DHCPDISCOVER on eth0 to 255.255.255.255 port 67 interval 7
DHCPDISCOVER on eth0 to 255.255.255.255 port 67 interval 9
DHCPDISCOVER on eth0 to 255.255.255.255 port 67 interval 14
DHCPDISCOVER on eth0 to 255.255.255.255 port 67 interval 9
DHCPDISCOVER on eth0 to 255.255.255.255 port 67 interval 9
DHCPDISCOVER on eth0 to 255.255.255.255 port 67 interval 8
DHCPDISCOVER on eth0 to 255.255.255.255 port 67 interval 5
No DHCPOFFERS received.
No working leases in persistent database - sleeping.
```

and my routes is:
```bash
tvbox:~#ip r
default via 192.168.1.1 dev wlan0 proto dhcp src 192.168.1.17 metric 4096
1.0.0.1 via 192.168.1.1 dev wlan0 proto dhcp src 192.168.1.17 metric 4096
1.1.1.1 via 192.168.1.1 dev wlan0 proto dhcp src 192.168.1.17 metric 4096
8.8.4.4 via 192.168.1.1 dev wlan0 proto dhcp src 192.168.1.17 metric 4096
8.8.8.8 via 192.168.1.1 dev wlan0 proto dhcp src 192.168.1.17 metric 4096
192.168.1.0/24 dev wlan0 proto kernel scope link src 192.168.1.17 metric 4096
192.168.1.1 dev wlan0 proto dhcp scope link src 192.168.1.17 metric 4096
```

```bash
tvbox:~#cat /sys/bus/mdio_bus/devices/*/{phy_interface,phy_id}|xargs
rgmii-txid internal 0x937c4032 0x01803301
```

**Reply by devmfc** [reply]
Yes, that is not a rtl8211F phy. But I think a JL2101. We can get that to work. 
Can you try the attached DTB instead? And then give me the same information again? (`dmesg` log and `cat /sys/bus/mdio_bus/devices/*/{phy_interface,phy_id}`)

[meson-sc2-ah212-generic-gbit.zip](https://github.com/devmfc/debian-on-amlogic/files/13461053/meson-sc2-ah212-generic-gbit.zip)

**Reply by NotDead** [reply]
```
rgmii 0x937c4032
```
[dmesg_new_dtb.log](https://github.com/devmfc/debian-on-amlogic/files/13462330/dmesg_new_dtb.log)

**Reply by devmfc** [reply]
Ok, thanx.

Add the lines below to boot.config. And uncomment one of the bootargs1 lines
at a time and reboot the box and see what happens.  
I hope one of these lines will fix the gigabit problem. When you know 
the best numbers, I will add them to a new DTB for that box.  
  
after each boot please gather the following information:  
output from `dmesg|grep meson8b-dwmac`   
output from `ethtool eth0`  
  
If this doesn't work I need the DTB from android and otherwise I'm out 
of ideas for now I'm afraid... 
  
~~~
##rgmii
#bootargs1=dwmac-meson8b.mc_val=0x1621 dwmac-meson8b.cali_val=0x10000
#bootargs1=dwmac-meson8b.mc_val=0x1621 dwmac-meson8b.cali_val=0x20000
#bootargs1=dwmac-meson8b.mc_val=0x1621 dwmac-meson8b.cali_val=0x30000
#bootargs1=dwmac-meson8b.mc_val=0x1621 dwmac-meson8b.cali_val=0x40000
#bootargs1=dwmac-meson8b.mc_val=0x1621 dwmac-meson8b.cali_val=0x50000
#bootargs1=dwmac-meson8b.mc_val=0x1621 dwmac-meson8b.cali_val=0x60000
#bootargs1=dwmac-meson8b.mc_val=0x1621 dwmac-meson8b.cali_val=0x70000
#bootargs1=dwmac-meson8b.mc_val=0x1621 dwmac-meson8b.cali_val=0x80000
#bootargs1=dwmac-meson8b.mc_val=0x1621 dwmac-meson8b.cali_val=0x90000
#bootargs1=dwmac-meson8b.mc_val=0x1621 dwmac-meson8b.cali_val=0xa0000
#bootargs1=dwmac-meson8b.mc_val=0x1621 dwmac-meson8b.cali_val=0xb0000
#bootargs1=dwmac-meson8b.mc_val=0x1621 dwmac-meson8b.cali_val=0xc0000
#bootargs1=dwmac-meson8b.mc_val=0x1621 dwmac-meson8b.cali_val=0xd0000
#bootargs1=dwmac-meson8b.mc_val=0x1621 dwmac-meson8b.cali_val=0xe0000
#bootargs1=dwmac-meson8b.mc_val=0x1621 dwmac-meson8b.cali_val=0xf0000
#
##rgmii-tx
#bootargs1=dwmac-meson8b.mc_val=0x1601 dwmac-meson8b.cali_val=0x10000
#bootargs1=dwmac-meson8b.mc_val=0x1601 dwmac-meson8b.cali_val=0x20000
#bootargs1=dwmac-meson8b.mc_val=0x1601 dwmac-meson8b.cali_val=0x30000
#bootargs1=dwmac-meson8b.mc_val=0x1601 dwmac-meson8b.cali_val=0x40000
#bootargs1=dwmac-meson8b.mc_val=0x1601 dwmac-meson8b.cali_val=0x50000
#bootargs1=dwmac-meson8b.mc_val=0x1601 dwmac-meson8b.cali_val=0x60000
#bootargs1=dwmac-meson8b.mc_val=0x1601 dwmac-meson8b.cali_val=0x70000
#bootargs1=dwmac-meson8b.mc_val=0x1601 dwmac-meson8b.cali_val=0x80000
#bootargs1=dwmac-meson8b.mc_val=0x1601 dwmac-meson8b.cali_val=0x90000
#bootargs1=dwmac-meson8b.mc_val=0x1601 dwmac-meson8b.cali_val=0xa0000
#bootargs1=dwmac-meson8b.mc_val=0x1601 dwmac-meson8b.cali_val=0xb0000
#bootargs1=dwmac-meson8b.mc_val=0x1601 dwmac-meson8b.cali_val=0xc0000
#bootargs1=dwmac-meson8b.mc_val=0x1601 dwmac-meson8b.cali_val=0xd0000
#bootargs1=dwmac-meson8b.mc_val=0x1601 dwmac-meson8b.cali_val=0xe0000
#bootargs1=dwmac-meson8b.mc_val=0x1601 dwmac-meson8b.cali_val=0xf0000
~~~

**Reply by NotDead** [reply]
Beforehand, I checked all the options with logging, made a short list where DHCP automatically triggered, now I will check without wi-fi, and everything that will work as it is, I will check on a clean OS.
[aml_network_logs.tar.gz](https://github.com/devmfc/debian-on-amlogic/files/13476938/aml_network_logs.tar.gz)

Thanks for help

**Reply by NotDead** [reply]
made dirtroom testing, all "rgmii" is non functional, now test "rgmii-tx"

**Reply by devmfc** [reply]
Do none of these settings work?? That's disappointing...

**Reply by NotDead** [reply]
Tested "rgmii-tx" - everything works from "1" to "c", tested with iperf3

preforms similar
```
##rgmii-tx
#bootargs1=dwmac-meson8b.mc_val=0x1601 dwmac-meson8b.cali_val=0x10000
#bootargs1=dwmac-meson8b.mc_val=0x1601 dwmac-meson8b.cali_val=0x20000
#bootargs1=dwmac-meson8b.mc_val=0x1601 dwmac-meson8b.cali_val=0x30000
#bootargs1=dwmac-meson8b.mc_val=0x1601 dwmac-meson8b.cali_val=0x40000
#bootargs1=dwmac-meson8b.mc_val=0x1601 dwmac-meson8b.cali_val=0x50000
#bootargs1=dwmac-meson8b.mc_val=0x1601 dwmac-meson8b.cali_val=0x60000
#bootargs1=dwmac-meson8b.mc_val=0x1601 dwmac-meson8b.cali_val=0x70000
#bootargs1=dwmac-meson8b.mc_val=0x1601 dwmac-meson8b.cali_val=0x80000
#bootargs1=dwmac-meson8b.mc_val=0x1601 dwmac-meson8b.cali_val=0x90000
#bootargs1=dwmac-meson8b.mc_val=0x1601 dwmac-meson8b.cali_val=0xa0000
#bootargs1=dwmac-meson8b.mc_val=0x1601 dwmac-meson8b.cali_val=0xb0000
#bootargs1=dwmac-meson8b.mc_val=0x1601 dwmac-meson8b.cali_val=0xc0000
```

**Reply by NotDead** [reply]
Hello, today I received another Vontar x4 tv box with a new hardware revision and it has everything working without "bootargs". The network performance remains at the same level

**Reply by NotDead** [reply]
I think we need 2 versions of the settings for different revisions? Here's a photo of the 2nd revision's insides

<details>
<summary>IMG</summary>


![20231126_141602](https://github.com/devmfc/debian-on-amlogic/assets/1535946/12f51a48-b48a-4785-9e77-5d39edf06c20)
</details>


It has RTL8211F

**Reply by devmfc** [reply]
Great, that is the PHY the dtb was made for. Why do you want to have does s905x4 boxes? Mainline support does not exist (yet), my images for the s905x4 (SC2) ar very experimental and quite slow. You would be better of with a s905x3 box or better still, a s922x box?

Anyway, attached is a modified DTB for the box with the JL2101 based on your tests, can you try that one (without the extra bootargs in boot.config)?

[meson-sc2-ah212-generic-gbit.zip](https://github.com/devmfc/debian-on-amlogic/files/13493701/meson-sc2-ah212-generic-gbit.zip)

**Reply by devmfc** [reply]
Maybe you can also test this one to see if we can get the vfd segment display to work

[meson-sc2-ah212-generic-gbit.zip](https://github.com/devmfc/debian-on-amlogic/files/13493970/meson-sc2-ah212-generic-gbit.zip)

**Reply by NotDead** [reply]
> Great, that is the PHY the dtb was made for. Why do you want to have does s905x4 boxes? Mainline support does not exist (yet), my images for the s905x4 (SC2) ar very experimental and quite slow. You would be better of with a s905x3 box or better still, a s922x box?

One of them is mine, I bought it to test av1 and maybe get some future-proof , and the other I bought on sale (for less than half price) to give as a gift. If x4 isn't that much different from x3, maybe we'll have mainline kernel support soon.

>Anyway, attached is a modified DTB for the box with the JL2101 based on your tests, can you try that one (without the extra bootargs in boot.config)?

Ok, I will test it, thanks for your time

>Maybe you can also test this one to see if we can get the vfd segment display to work

and that too

**Reply by NotDead** [reply]
>Anyway, attached is a modified DTB for the box with the JL2101 based on your tests, can you try that one (without the extra bootargs in boot.config)?

works great!

>Maybe you can also test this one to see if we can get the vfd segment display to work

shows me the time, temp, etc, but without the network =)

**Reply by devmfc** [reply]
Does the second one also not work with the other box with RTL8211F?
Attached is a combined dtb of the two. You have to either change the box-config to use this DTB or rename the dtb to meson-sc2-ah212-generic-gbit.dtb

[meson-sc2-vontarx4-gbit.zip](https://github.com/devmfc/debian-on-amlogic/files/13506637/meson-sc2-vontarx4-gbit.zip)

**Reply by NotDead** [reply]
>Does the second one also not work with the other box with RTL8211F?

yes it works great

>Attached is a combined dtb of the two.

works on RTL8211F and JL2101 


I took a closer look at the display and realized that it shows at least something - but it's not clear what. Time does not coincide with the time zone in OS (1 hour difference, in OS CET timezone - on display is CET -1), the temperature shows not clear what, there is 1 digit and a sign celsius, then in rotation 52 and lit wi-fi icon, and in general there are icons light up on an incomprehensible to me principle

**Reply by devmfc** [reply]
Great, I will add the new DTB and new box config in the next release.

I think the led mapping is not ok. It should rotate between 
- time with a blinking colon
- temperature℃
- =last digits ip address

If you want / have the time to do some puzzling to get the mapping right: see /usr/lib/tm16xx-display/tm16xx-map.sh for examples.
And maybe I have the IC wrong? You opened the box right? Is it a FD650 or FD6551 led controller? 

If you just want to turn it of: `echo 0 > /sys/bus/platform/drivers/tm16xx/*/brightness`

**Comment by jojkol** [comment]
> 您好。感谢您的出色工作！
> 
> 我

> 您好。记得您的出色工作！
> 
> 我在 X96 X4 千兆电视盒上兼容 debian 和 ubuntu，配有 4GB RAM。但是，网络不工作。我将 LAN 电缆插入盒子，什么发生。
> 
> 您可能解释一下如何让这个盒子上的网络工作吗？

Buddy, have you solved the network problem? I have tried many times, but still can't connect to the network. As long as the box is in Debian system, my router background management cannot display this device. Only when the box is in Android system, the network is normal.

**Reply by devmfc** [reply]
Do you have a 100MB ethernet? If so, sorry, will not work at this moment, only some gigabit PHY's will work on an S905X4..
